openbas-api/src/test/java/io/openbas/service/ExerciseServiceTest.java

```diff
@@ -1,81 +1,138 @@
-            }
-        });
-    }
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class ExerciseServiceTest {
+
+  @Mock
+  GrantService grantService;
+  @Mock
+  InjectService injectService;
+  @Mock
+  InjectDuplicateService injectDuplicateService;
+  @Autowired
+  private TeamService teamService;
+  @Mock
+  VariableService variableService;
+
+  @Autowired
+  private ArticleRepository articleRepository;
+  @Autowired
+  private ExerciseRepository exerciseRepository;
+  @Autowired
+  private TeamRepository teamRepository;
+
+  @Autowired
+  private UserRepository userRepository;
+  @Autowired
+  private InjectRepository injectRepository;
+  @Autowired
+  private ExerciseTeamUserRepository exerciseTeamUserRepository;
+  @Autowired
+  private InjectorContractRepository injectorContractRepository;
+
+  private static String USER_ID;
+  private static String TEAM_ID;
+  private static String INJECT_ID;
+
+  @InjectMocks
+  private ExerciseService exerciseService;
+
+  @BeforeEach
+  void setUp() {
+    exerciseService = new ExerciseService(grantService, injectService, injectDuplicateService,
+        teamService, variableService, articleRepository, exerciseRepository, teamRepository,
+        exerciseTeamUserRepository, injectRepository);
+  }
+
+  @AfterAll
+  public void teardown() {
+    this.userRepository.deleteById(USER_ID);
+    this.teamRepository.deleteById(TEAM_ID);
+    this.injectRepository.deleteById(INJECT_ID);
+  }
+
+  @DisplayName("Should create new contextual teams while exercise duplication")
+  @Test
+  @Transactional(rollbackFor = Exception.class)
+  void createNewContextualTeamsWhileExerciseDuplication() {
+    // -- PREPARE --
+    List<Team> exerciseTeams = new ArrayList<>();
+    Team contextualTeam = this.teamRepository.save(getTeam(null, "fakeTeamName1", true));
+    exerciseTeams.add(contextualTeam);
+    Team noContextualTeam = this.teamRepository.save(getTeam(null, "fakeTeamName2", false));
+    exerciseTeams.add(noContextualTeam);
+    Exercise exercise = this.exerciseRepository.save(getExercise(exerciseTeams));
+
+    // -- EXECUTE --
+    Exercise exerciseDuplicated = exerciseService.getDuplicateExercise(exercise.getId());
+
+    // -- ASSERT --
+    assertNotEquals(exercise.getId(), exerciseDuplicated.getId());
+    assertEquals(2, exerciseDuplicated.getTeams().size());
+    exerciseDuplicated.getTeams().forEach(team -> {
+      if (team.getContextual()) {
+        assertNotEquals(contextualTeam.getId(), team.getId());
+        assertEquals(contextualTeam.getName(), team.getName());
+      } else {
+        assertEquals(noContextualTeam.getId(), team.getId());
+      }
+    });
+  }
+
+  @DisplayName("Should remove team from exercise")
+  @Test
+  void testRemoveTeams() {
+    // -- PREPARE --
+    User user = getUser();
+    User userSaved = this.userRepository.saveAndFlush(user);
+    USER_ID = userSaved.getId();
+    Team team = getTeam(user);
```

Comment:
should be `Team team = getTeam(userSaved);`  ?

Reply:
Good catch